ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/commitmarkable/FailAfterPrepareBase.java

```diff
@@ -50,6 +52,12 @@
     private SimpleXAResource xaResource;
     private CommitMarkableResourceRecordRecoveryModule recoveryModule;
 
+    @Before
+    public void setUp() {
+        // not running with JDK6
+        Assume.assumeTrue(!System.getProperty("java.version").startsWith("1.6"));
```

Comment:
Why do we only run the tests on JDK 1.6?

Reply:
It's the test for not being run on JDK 1.6 as there is the trouble as test checks for suppressed exceptions. Skipping these few tests on 1.6 was the shortcut to proceed with testing on JDK6.

Reply:
Will we run these tests for JDK7+ still?

Reply:
Is that because we aren't allowed to do introspection to see if suppressed exceptions are available (I think that is what the later code did)? And is there no check for 1.5 because we don't support 1.5?

Reply:
@tomjenkinson  Yes. It will be still running on JDK7+.
@mmusgrov It's because the tests assertion work with the suppressed exceptions. I would need to change the logic of the tests which I don't want to do.
Yes, there is not check for 1.5 as it's not possible to start Narayana on JDK1.5 (aka. we do not support it).

Reply:
I meant why aren't we using refection to see if the method exists like we do in the main code (
https://github.com/jbosstm/narayana/blob/4.17.43.Final/ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/transaction/arjunacore/TransactionImple.java#L1684) rather than checking for specific JDK versions.

Reply:
By my opinion we can use both approaches here. The reflection targets the difference more precisely as we do use from JDK API only this method.

Reply:
And the reason why I chose to use the check for the java version is that it sounds me more reasonable for the test approach.

Reply:
I will use the similar approach as for TransactionImple to be done in the same way